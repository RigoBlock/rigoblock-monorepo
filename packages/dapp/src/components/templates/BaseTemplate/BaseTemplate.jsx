import './BaseTemplate.scss'
import { connect } from 'react-redux'
import LeftNavbar from '../../organisms/LeftNavbar'
import Modal from '../../molecules/Modal'
import PropTypes from 'prop-types'
import React, { PureComponent } from 'react'
import TopNavbar from '../../organisms/TopNavbar'
import classNames from 'classnames'
import globalActions from '../../../actions/global-actions'

class BaseTemplate extends PureComponent {
  render() {
    const { children, className, modalComponent, dispatch } = this.props
    const contentClass = classNames('page-main-content', className)
    return (
      <div>
        <TopNavbar />
        <div className="page-content-section">
          <LeftNavbar />
          <div className={contentClass}>{children}</div>
        </div>
        <Modal
          Component={modalComponent}
          onClickOutside={() => dispatch(globalActions.closeModal())}
        />
      </div>
    )
  }
}

BaseTemplate.propTypes = {
  children: PropTypes.node.isRequired,
  dispatch: PropTypes.func.isRequired,
  className: PropTypes.string,
  modalComponent: PropTypes.element
}

BaseTemplate.defaultProps = {
  className: undefined, // TODO use null
  modalComponent: null
}

BaseTemplate = connect(state => {
  const { modalComponent } = state.globalReducer
  return { modalComponent }
})(BaseTemplate)

export default BaseTemplate
