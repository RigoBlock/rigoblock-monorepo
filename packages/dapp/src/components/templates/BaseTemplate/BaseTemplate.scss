@import "../../_settings/variables.scss";

.content-section {
  display: flex;
  flex-direction: row;
  width: 100vw;
  height: calc(100% - $dist-l);
}

.content {
  width: 85vw;
  margin-right: $dist-s;
}
