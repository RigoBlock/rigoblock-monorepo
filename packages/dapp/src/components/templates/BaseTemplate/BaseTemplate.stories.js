import '../../_settings/_base.scss'
import * as ROUTES from '../../../constants/routes'
import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import { select, withKnobs } from '@storybook/addon-knobs/react'
import { storiesOf } from '@storybook/react'
import BaseTemplate from './BaseTemplate'
import React from 'react'
import mockStore, { defaultState } from '../../../fixtures/store'

const getState = () => ({
  ...defaultState,
  ...{
    routing: {
      location: {
        pathname: select('url', ROUTES, ROUTES.DASHBOARD)
      }
    }
  }
})

const store = mockStore({ getState })

storiesOf('Templates/BaseTemplate', module)
  .addDecorator(withKnobs)
  .addDecorator(story => (
    <Provider store={store}>
      <MemoryRouter>{story()}</MemoryRouter>
    </Provider>
  ))
  .add('default', () => <BaseTemplate>Base template children</BaseTemplate>)
