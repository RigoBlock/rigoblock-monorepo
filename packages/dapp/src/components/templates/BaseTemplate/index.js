import BaseTemplate from './BaseTemplate'

export default BaseTemplate
