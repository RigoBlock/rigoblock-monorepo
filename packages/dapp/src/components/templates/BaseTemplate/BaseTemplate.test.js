import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import BaseTemplate from './BaseTemplate.jsx'

describe('BaseTemplate component', () => {
  it('renders correctly', () => {
    expect(
      toJson(shallow(<BaseTemplate>Test Content</BaseTemplate>))
    ).toMatchSnapshot()
  })
})
