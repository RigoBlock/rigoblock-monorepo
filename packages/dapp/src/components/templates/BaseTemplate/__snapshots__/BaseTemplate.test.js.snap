// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseTemplate component renders correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [MockFunction],
      "getState": [Function],
      "subscribe": [MockFunction],
    }
  }
>
  <MemoryRouter
    initialEntries={
      Array [
        Object {
          "key": "testKey",
        },
      ]
    }
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "testKey",
              "pathname": "/",
              "search": "",
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(BaseTemplate)>
        <BaseTemplate
          closeModal={[Function]}
          init={[Function]}
          modalComponent={null}
          openModal={[Function]}
        >
          <div>
            <TopNavbar>
              <div
                className="top-navbar"
              >
                <Link
                  className="navbar-title"
                  size="medium"
                  to="/"
                >
                  <Link
                    className="link medium navbar-title"
                    replace={false}
                    to="/"
                  >
                    <a
                      className="link medium navbar-title"
                      href="/"
                      onClick={[Function]}
                    >
                      RigoBlock
                    </a>
                  </Link>
                </Link>
                <UserMenu>
                  <div
                    className="user-menu"
                  >
                    <Link
                      className={null}
                      size="medium"
                      to="/preferences/"
                    >
                      <Link
                        className="link medium"
                        replace={false}
                        to="/preferences/"
                      >
                        <a
                          className="link medium"
                          href="/preferences/"
                          onClick={[Function]}
                        >
                          Preferences
                        </a>
                      </Link>
                    </Link>
                    <Link
                      className={null}
                      size="medium"
                      to="/help/"
                    >
                      <Link
                        className="link medium"
                        replace={false}
                        to="/help/"
                      >
                        <a
                          className="link medium"
                          href="/help/"
                          onClick={[Function]}
                        >
                          Help
                        </a>
                      </Link>
                    </Link>
                  </div>
                </UserMenu>
              </div>
            </TopNavbar>
            <div
              className="page-content-section"
            >
              <LeftNavbar>
                <div
                  className="left-navbar"
                >
                  <WrapperWithDivider
                    Divider={[Function]}
                  >
                    <div
                      className="dividing-wrapper"
                    >
                      <div
                        key="wrap-div-0"
                      >
                        <Connect(AccountView)>
                          <AccountView
                            accounts={
                              Object {
                                "0x242B2Dd21e7E1a2b2516d0A3a06b58e2D9BF9196": Object {
                                  "balance": "64905172851266399404",
                                  "provider": "metamask",
                                  "vaultTransactions": Object {
                                    "0x86a1ba4d485ce346bded508e2426798f825558be": Object {
                                      "0x2c3cdebdba5271d54780f3368f90982582b734610175e620c8b964382de9f351": Object {
                                        "account": "0x242b2dd21e7e1a2b2516d0a3a06b58e2d9bf9196",
                                        "date": 1531227188000,
                                        "symbol": "ETH",
                                        "type": "Deposit",
                                        "units": "23860000",
                                        "value": "23860000000000000000",
                                      },
                                      "0xc8c775cec1666c7bd55eb60956dc2aad39beea0846396c051410838c74c6d37c": Object {
                                        "account": "0x242b2dd21e7e1a2b2516d0a3a06b58e2d9bf9196",
                                        "date": 1531227188000,
                                        "symbol": "ETH",
                                        "type": "Withdraw",
                                        "units": "1000000",
                                        "value": "1000000000000000000",
                                      },
                                    },
                                  },
                                  "vaults": Object {
                                    "0x421e1cef6e85e78da2470e54af64a626f45afb85": Object {
                                      "group": "0x7ce6e371085cb611fb46d5065397223ef2f952ff",
                                      "id": "1",
                                      "name": "Third Vault",
                                      "owner": "0x7328ef1d7ab7583eb9968b2f4a9c900f8a2e2d6d",
                                      "symbol": "das",
                                      "totalSupply": "21343170000000000000",
                                      "transactionFee": "2.5",
                                    },
                                    "0x86a1ba4d485ce346bded508e2426798f825558be": Object {
                                      "group": "0x7ce6e371085cb611fb46d5065397223ef2f952ff",
                                      "id": "0",
                                      "name": "First Vault",
                                      "owner": "0x242b2dd21e7e1a2b2516d0a3a06b58e2d9bf9196",
                                      "symbol": "asd",
                                      "totalSupply": "28640000000000000000",
                                      "transactionFee": "2.5",
                                    },
                                  },
                                },
                                "0x7328ef1d7ab7583eb9968b2f4a9c900f8a2e2d6d": Object {
                                  "balance": "67999999999977505798",
                                  "provider": "metamask",
                                },
                                "0x8bb7481495d45ccd5cffae1c3a84155fea85a323": Object {
                                  "balance": "87999999999999410580",
                                  "provider": "metamask",
                                },
                              }
                            }
                            currentAccount="0x242B2Dd21e7E1a2b2516d0A3a06b58e2D9BF9196"
                            dispatch={[MockFunction]}
                          >
                            <div
                              className="account-view"
                            >
                              <Title
                                className={null}
                                size="small"
                              >
                                <div
                                  className="title small"
                                >
                                  Accounts
                                </div>
                              </Title>
                              <List
                                Component={[Function]}
                                Divider={null}
                                className="account-list"
                                data={
                                  Array [
                                    Object {
                                      "number": "0x242B2Dd21e7E1a2b2516d0A3a06b58e2D9BF9196",
                                      "provider": "metamask",
                                    },
                                    Object {
                                      "number": "0x7328ef1d7ab7583eb9968b2f4a9c900f8a2e2d6d",
                                      "provider": "metamask",
                                    },
                                    Object {
                                      "number": "0x8bb7481495d45ccd5cffae1c3a84155fea85a323",
                                      "provider": "metamask",
                                    },
                                  ]
                                }
                              >
                                <div
                                  className="account-list"
                                >
                                  <AccountItem
                                    appearance="short"
                                    key="0"
                                    number="0x242B2Dd21e7E1a2b2516d0A3a06b58e2D9BF9196"
                                    provider="metamask"
                                  >
                                    <div
                                      className="account-item-short"
                                    >
                                      <div
                                        className="logo"
                                      >
                                        <Component>
                                          SVG image metamask-fox.svg
                                        </Component>
                                      </div>
                                      <div
                                        className="info"
                                      >
                                        <span
                                          className="provider"
                                        >
                                          MetaMask
                                        </span>
                                        <span
                                          className="account-number"
                                        >
                                          **** 9196
                                        </span>
                                      </div>
                                    </div>
                                  </AccountItem>
                                  <AccountItem
                                    appearance="short"
                                    key="1"
                                    number="0x7328ef1d7ab7583eb9968b2f4a9c900f8a2e2d6d"
                                    provider="metamask"
                                  >
                                    <div
                                      className="account-item-short"
                                    >
                                      <div
                                        className="logo"
                                      >
                                        <Component>
                                          SVG image metamask-fox.svg
                                        </Component>
                                      </div>
                                      <div
                                        className="info"
                                      >
                                        <span
                                          className="provider"
                                        >
                                          MetaMask
                                        </span>
                                        <span
                                          className="account-number"
                                        >
                                          **** 2d6d
                                        </span>
                                      </div>
                                    </div>
                                  </AccountItem>
                                  <AccountItem
                                    appearance="short"
                                    key="2"
                                    number="0x8bb7481495d45ccd5cffae1c3a84155fea85a323"
                                    provider="metamask"
                                  >
                                    <div
                                      className="account-item-short"
                                    >
                                      <div
                                        className="logo"
                                      >
                                        <Component>
                                          SVG image metamask-fox.svg
                                        </Component>
                                      </div>
                                      <div
                                        className="info"
                                      >
                                        <span
                                          className="provider"
                                        >
                                          MetaMask
                                        </span>
                                        <span
                                          className="account-number"
                                        >
                                          **** a323
                                        </span>
                                      </div>
                                    </div>
                                  </AccountItem>
                                </div>
                              </List>
                            </div>
                          </AccountView>
                        </Connect(AccountView)>
                        <divider>
                          <div
                            className="left-navbar-divider"
                          />
                        </divider>
                      </div>
                      <div
                        key="wrap-div-1"
                      >
                        <Connect(NavigationView)>
                          <NavigationView
                            currentUrl="/vaults/0"
                            dispatch={[MockFunction]}
                          >
                            <div>
                              <List
                                Component={[Function]}
                                Divider={null}
                                className="navigation-view"
                                data={
                                  Array [
                                    Object {
                                      "className": "",
                                      "icon": "dashboard",
                                      "link": Object {
                                        "text": "Dashboard",
                                        "to": "/",
                                      },
                                    },
                                    Object {
                                      "className": "",
                                      "icon": "account_balance_wallet",
                                      "link": Object {
                                        "text": "Dragos",
                                        "to": "/dragos/",
                                      },
                                    },
                                    Object {
                                      "className": "active",
                                      "icon": "lock",
                                      "link": Object {
                                        "text": "Vaults",
                                        "to": "/vaults/",
                                      },
                                    },
                                  ]
                                }
                              >
                                <div
                                  className="navigation-view"
                                >
                                  <ViewLink
                                    className=""
                                    icon="dashboard"
                                    key="0"
                                    link={
                                      Object {
                                        "text": "Dashboard",
                                        "to": "/",
                                      }
                                    }
                                  >
                                    <Link
                                      className="view-link"
                                      size="small"
                                      to="/"
                                    >
                                      <Link
                                        className="link small view-link"
                                        replace={false}
                                        to="/"
                                      >
                                        <a
                                          className="link small view-link"
                                          href="/"
                                          onClick={[Function]}
                                        >
                                          <Icon
                                            className=""
                                            onClick={[Function]}
                                            size="medium"
                                            tooltipText=""
                                            type="dashboard"
                                          >
                                            <div
                                              className="material-icons icon medium"
                                              onClick={[Function]}
                                            >
                                              dashboard
                                            </div>
                                          </Icon>
                                          <span>
                                            Dashboard
                                          </span>
                                        </a>
                                      </Link>
                                    </Link>
                                  </ViewLink>
                                  <ViewLink
                                    className=""
                                    icon="account_balance_wallet"
                                    key="1"
                                    link={
                                      Object {
                                        "text": "Dragos",
                                        "to": "/dragos/",
                                      }
                                    }
                                  >
                                    <Link
                                      className="view-link"
                                      size="small"
                                      to="/dragos/"
                                    >
                                      <Link
                                        className="link small view-link"
                                        replace={false}
                                        to="/dragos/"
                                      >
                                        <a
                                          className="link small view-link"
                                          href="/dragos/"
                                          onClick={[Function]}
                                        >
                                          <Icon
                                            className=""
                                            onClick={[Function]}
                                            size="medium"
                                            tooltipText=""
                                            type="account_balance_wallet"
                                          >
                                            <div
                                              className="material-icons icon medium"
                                              onClick={[Function]}
                                            >
                                              account_balance_wallet
                                            </div>
                                          </Icon>
                                          <span>
                                            Dragos
                                          </span>
                                        </a>
                                      </Link>
                                    </Link>
                                  </ViewLink>
                                  <ViewLink
                                    className="active"
                                    icon="lock"
                                    key="2"
                                    link={
                                      Object {
                                        "text": "Vaults",
                                        "to": "/vaults/",
                                      }
                                    }
                                  >
                                    <Link
                                      className="view-link active"
                                      size="small"
                                      to="/vaults/"
                                    >
                                      <Link
                                        className="link small view-link active"
                                        replace={false}
                                        to="/vaults/"
                                      >
                                        <a
                                          className="link small view-link active"
                                          href="/vaults/"
                                          onClick={[Function]}
                                        >
                                          <Icon
                                            className=""
                                            onClick={[Function]}
                                            size="medium"
                                            tooltipText=""
                                            type="lock"
                                          >
                                            <div
                                              className="material-icons icon medium"
                                              onClick={[Function]}
                                            >
                                              lock
                                            </div>
                                          </Icon>
                                          <span>
                                            Vaults
                                          </span>
                                        </a>
                                      </Link>
                                    </Link>
                                  </ViewLink>
                                </div>
                              </List>
                            </div>
                          </NavigationView>
                        </Connect(NavigationView)>
                      </div>
                    </div>
                  </WrapperWithDivider>
                </div>
              </LeftNavbar>
              <div
                className="page-main-content"
              >
                Test Content
              </div>
            </div>
            <Modal
              Component={null}
              onClickOutside={[Function]}
            />
          </div>
        </BaseTemplate>
      </Connect(BaseTemplate)>
    </Router>
  </MemoryRouter>
</Provider>
`;
